heroic-component/src/main/java/com/spotify/heroic/metric/QueryResult.java

```diff
@@ -95,8 +114,51 @@
                 limits.add(ResultLimit.GROUP);
             }
 
-            return new QueryResult(range, groupLimit.limitList(groups), errors, trace,
-                new ResultLimits(limits.build()));
+            QueryResult queryResult = new QueryResult(range, groupLimit.limitList(groups), errors,
+                                                      trace, new ResultLimits(limits.build()));
+
+            queryResult.logQueryInfo(request, requestMetadata,
+                                     logQueries, logQueriesThresholdDataPoints);
+
+            return queryResult;
         };
     }
+
+    public void logQueryInfo(final FullQuery.Request request,
```

Comment:
This does not make sense to live in `QueryResult`. Move it to an interface (e.g. `QueryLogger`) and provide an implementation to it in core. Then make use of it in CoreQueryManager.

Similarly do not use static fields for `totalQueriesProcessed` and `queriesAboveThreshold`. When they live outside of `QueryResult`, this is not a problem.

Similarly for `queryLog`, which would be a member of the `QueryLogger` implementation.